### ISSUE TITLE
info on validation methods for TFT

### ISSUE THREAD

**Issue by ianux22** [jdb78/pytorch-forecasting#1283]
Hello everyone! I'm new to the library and I'm trying to understand how to approach the validation part.
I want to understand how to run those 2 approaches: 

- Let's say I want to use a validation set which is not used for the training (like a hold-out set).

In this case, looking at the [TFT tutorial](https://pytorch-forecasting.readthedocs.io/en/stable/tutorials/stallion.html), I can use "the last max_prediction_length points in time" as validation set.

```
training = TimeSeriesDataSet(
    data[lambda x: x.time_idx <= training_cutoff],
           .............
            )

# create validation set (predict=True) which means to predict the last max_prediction_length points in time
# for each series
validation = TimeSeriesDataSet.from_dataset(training, data, predict=True, stop_randomization=True)

# create dataloaders for model
batch_size = 128  # set this between 32 to 128
train_dataloader = training.to_dataloader(train=True, batch_size=batch_size, num_workers=0)
val_dataloader = validation.to_dataloader(train=False, batch_size=batch_size * 10, num_workers=0)
```
Doing something like above would result in using the last max_prediction_length for validation and not for the training, correct? 

- Second approach. Let's say I want a forward chaining cross validation. To be clear, here an example. There is an initial amount of data to train on, let's say five folds in this example, and then you evaluate on the sixth fold and save that performance metric. You re-train now on the first six folds and evaluate on the seventh. You repeat until all folds are exhausted and again take the average of your performance metric. The folds using this technique would look like this:

![image](https://user-images.githubusercontent.com/72081574/229211556-a7beb0f2-4be7-43b9-ab71-7690842b2f2d.png)

To run something like this, I assume I need to combine train and validation together and use a code like this, correect? 

```
training = TimeSeriesDataSet(
    data[lambda x: x.time_idx <= training_cutoff],
           .............
            )

# create validation set (predict=True) which means to predict the last max_prediction_length points in time
# for each series
validation = TimeSeriesDataSet.from_dataset(training, data, predict=False, stop_randomization=True)

# create dataloaders for model
batch_size = 128  # set this between 32 to 128
train_dataloader = training.to_dataloader(train=True, batch_size=batch_size, num_workers=0)
val_dataloader = validation.to_dataloader(train=False, batch_size=batch_size * 10, num_workers=0)
```

Is my approach correct for both the methologies or am I missing something?

**Comment by grosestq** [comment]
Hey,

in my opinion:

if you really want to increase the training set each time, you have to build separate models (e.g. with a loop).
